docs/languages/csharp/expression-trees-building.md

```diff
@@ -0,0 +1,218 @@
+var addition = Expression.Add(one, two);
+```
+
+Once you've got the addition expression, you can create the lambda
+expression:
+
+```cs
+var lamdba = Expression.Lambda(addition);
+```
+
+This is a very simple LambdaExpression, because it contains no arguments.
+Later in this section, you'll see how to map arguments to parameters
+and build more complicated expressions.
+
+For expressions that are as simple as this one, you may combine all the
+calls into a single statement:
+
+```cs
+var lambda = Expression.Lambda(
+    Expression.Add(
+        Expression.Constant(1, typeof(int)),
+        Expression.Constant(2, typeof(int))
+    )
+);
+```
+
+## Building a Tree
+
+That's the basics of building an expression tree in memory. More
+complex trees generally mean more node types, and more nodes in the
+tree. Let's run through one more example and show two more node types
+that you will typically build when you create expression trees:
+the argument nodes, and method call nodes.
+
+Let's build an expression tree to create this expression:
+
+```cs
+Expression<Func<double, double, double>> distanceCalc =
+    (x, y) => Math.Sqrt(x * x + y * y);
+```
+ 
+You'll start by creating parameter expressions for `x` and `y`:
+
+```cs
+var xParameter = Expression.Parameter(typeof(double), "x");
+var yParameter = Expression.Parameter(typeof(double), "y");
+```
+
+Creating the multiplication and addition expressions follows the pattern
+you've already seen:
+
+```cs
+var xSquared = Expression.Multiply(xParameter, xParameter);
+var ySquared = Expression.Multiply(yParameter, yParameter);
+var sum = Expression.Add(xSquared, ySquared);
+```
+
+Next, you need to create a method call expression for the call to
+`Math.Sqrt`.  This requires creating an expression that calls
+`Math.Sqrt`, and then updating that expression so that the
+arguments represent the sum of xSquared and ySquared:
+
+```cs
+Expression<Func<double, double>> sqrt = (x) => Math.Sqrt(x);
+var methodCall = sqrt.Body as MethodCallExpression;
+var distance = methodCall.Update(default(Expression), new List<Expression> { sum });
+```
+
+> Those familiar with the Reflection APIs may wonder why I didn't use
+> the Reflection APIs to retrieve the `MethodInfo` object for the `Math.Sqrt`
+> method. The answer is simple: that method is not available on .NET Core. 
+
+And  then finally, you put the method call into a lambda expression,
+and make sure to define the arguments to the lambda expression:
+
+```cs
+var distanceLambda = Expression.Lambda(
+    distance,
+    xParameter,
+    yParameter);
+```
+
+In this more complicated example, you see a couple more techniques that
+you will often need to create expression trees.
+
+First, you need to create the objects that represent parameters or
+local variables before you use them. Once you've created those objects,
+you can use them in your expression tree wherever you need.
+
+Second, you need to use a subset of the Reflection APIs to create a `MethodInfo` object
```

Comment:
Same feedback as above.

What subset are you referring to?  When building and modifying the `sqrt` expression you're casting it to the derived type and calling the `Update` method without any reflection calls (e.g. `GetType()`).